library/aesce.c

```diff
@@ -199,6 +199,7 @@ static uint8x16_t aesce_encrypt_block(uint8x16_t block,
 /* Two rounds of AESCE decryption */
 #define AESCE_DECRYPT_ROUND_X2        AESCE_DECRYPT_ROUND; AESCE_DECRYPT_ROUND
 
+#if !defined(MBEDTLS_BLOCK_CIPHER_NO_DECRYPT)
```

Comment:
Here we can drop the guard and just use `MBEDTLS_MAYBE_UNUSED` as this is static, so the compiler will remove it (I tested this, no impact on code size).

Reply:
> If a static function / variable is guarded by a single/simple macro, should we guard it directly or use MBEDTLS_MAYBE_UNUSED to let compiler remove it for us?

The compiler would remove the unused static functions and variables. But in your case, I think it is a compiler error before the removing happens. The preprocessor runs before the compiler, and it removes `mbedtls_aes_setkey_dec` since it is guarded by preprocessor macro. So you need to guard the function call.

Reply:
No strong preference on this one. We use `MBEDTLS_MAYBE_UNUSED` in aes.c, because the guards got so complicated I wanted to remove as many as possible and let the compiler remove functions automatically as far as possible. But `aesce.c` doesn't have this problem so either approach is OK in my view.